libraries/mbed/targets/hal/TARGET_NXP/TARGET_LPC81X/pinmap.c

```diff
@@ -27,7 +27,17 @@ __IO uint32_t* IOCON_REGISTERS[18] = {
 };
 
 void pin_function(PinName pin, int function) {
-    
+    if (function == 0) {
+        // Disable initial fixed function for P0_2, P0_3 and P0_5
+        uint32_t enable = 0;
+        if (pin == P0_2)
+            enable = 1 << 3;
+        else if (pin == P0_3)
+            enable = 1 << 2;
+        else if (pin == P0_5)
+            enable = 1 << 6;
+        LPC_SWM->PINENABLE0 |= enable;
+    }
```

Comment:
The LPC81X family does not have a conventional list of functions per pin, but instead each pin can received any of the "movable" functions. The only exceptions are known as "fixed-pin functions" and are nEnabled using the PINENABLE0 register (1 == disabled). On boot 3 of these fixed functions are enabled:
- SWDIO_EN is set on P0_2
- SWCLK_EN is set on P0_3
- RESET_EN is set on P0_5
If you want to use one of those pins as GPIO, you must first disable the fixed-pin function, and they'll return to their default GPIO function.
However you usually don't want to disable the RESET or the SW functions, but on a LPC810 package that has only 8 pins, this is more than appreciable.